service.languagepreferencemanager/resources/lib/prefutils.py

```diff
@@ -0,0 +1,369 @@
+                                    log(LOG_INFO,'CondSubs : one subtitle track is found matching Keyword Blacklist : {0}. Skipping it.'.format(','.join(settings.keyword_blacklist)))
+                                    continue
+                                if (settings.ignore_signs_on and self.isSignsSub(sub['name'])):
+                                    log(LOG_INFO,'CondSubs : ignore_signs toggle is on and one such subtitle track is found. Skipping it.')
+                                    continue
+                                if ((sub_code == sub['language']) or (sub_name == sub['language'])):
+                                    if (self.testForcedFlag(forced, sub['name'])):
+                                        log(LOG_INFO, 'Language of subtitle {0} matches conditional preference {1} ({2}:{3}) forced {4}'.format((sub['index']+1), i, audio_name, sub_name, forced) )
+                                        return sub['index']
+                            log(LOG_INFO, 'Conditional subtitle: no match found for preference {0} ({1}:{2})'.format(i, audio_name, sub_name) )
+                i += 1
+        return -2
+
+    def isSignsSub(self, subName):
+        test = subName.lower()
+        matches = ['signs']
+        return any(x in test for x in matches)
+    
+    def testForcedFlag(self, forced, subName):
+        test = subName.lower()
+        matches = ['forced', 'forcés']
+        found = any(x in test for x in matches)
+        return ((forced == 'false') and not found) or ((forced == 'true') and found)
+
+    def isExternalSub(self, subName):
+        test = subName.lower()
+        matches = ['ext']
+        return any(x in test for x in matches)
+    
+    def getDetails(self):
+        activePlayers ='{"jsonrpc": "2.0", "method": "Player.GetActivePlayers", "id": 1}'
+        json_query = xbmc.executeJSONRPC(activePlayers)
+        #json_query = unicode(json_query, 'utf-8', errors='ignore')
+        json_response = simplejson.loads(json_query)
+        activePlayerID = json_response['result'][0]['playerid']
+        details_query_dict = {  "jsonrpc": "2.0",
+                                "method": "Player.GetProperties",
+                                "params": { "properties": 
+                                            ["currentaudiostream", "audiostreams", "subtitleenabled",
+                                             "currentsubtitle", "subtitles" ],
+                                            "playerid": activePlayerID },
+                                "id": 1}
+        details_query_string = simplejson.dumps(details_query_dict)
+        json_query = xbmc.executeJSONRPC(details_query_string)
+        #json_query = unicode(json_query, 'utf-8', errors='ignore')
+        json_response = simplejson.loads(json_query)
+        
+        if 'result' in json_response and json_response['result'] != None:
```

Comment:
`json_response['result'] is not None`

Reply:
Not mandatory hence will be fixed in next update.